common/hash_data_other.go

```diff
@@ -0,0 +1,70 @@
+//go:build !windows
+// +build !windows
+
+package common
+
+import (
+	"encoding/json"
+	"io"
+	"os"
+	"path/filepath"
+	"time"
+)
+
+func TryGetHashData(fullpath string) (SyncHashData, error) {
+	// get meta file name
+	dir, fn := filepath.Split(fullpath)
```

Comment:
Using file path we are creating a path to store our hash. This code is duplicate in Get and Set methods. Shall we unify this so that later in case we need to make any change to this logic it's easy to maintain.

Reply:
Sure.